addons/mrp/models/mrp_production.py

```diff
@@ -2111,7 +2111,7 @@ def _check_immediate(self):
         pd = self.env['decimal.precision'].precision_get('Product Unit of Measure')
         for production in self:
             if all(float_is_zero(ml.qty_done, precision_digits=pd) for
-                    ml in production.move_raw_ids.move_line_ids.filtered(lambda m: m.state not in ('done', 'cancel'))
+                    ml in production.move_raw_ids.move_line_ids.filtered(lambda m: m.state not in ('done', 'cancel') and not m.qty_done > 0)
```

Comment:
This part of the condition becomes useless
The filtering only keeps the SML with a `qty_done` equal to zero (can't be negative). Then you check that the `qty_done` of the kept SMLs is zero -> this will always be the case